llvm/lib/Target/AMDGPU/AMDGPUSplitModule.cpp

```diff
@@ -256,11 +256,12 @@ calculateFunctionCosts(SplitModuleLogger &SML, GetTTIFn GetTTI, Module &M,
   }
 
   CostType FnCost = (ModuleCost - KernelCost);
+  CostType ModuleCostOr1 = ModuleCost ? ModuleCost : 1;
```

Comment:
Detecting this earlier than just early-retun the pass would be better but this works as a temporary fix.
The pass is going to be rewritten very soon anyway, I'll make sure this is handled in the rewrite.

Reply:
Ah right, thank you.

To be honest I'm not sure how the pass is supposed to work in general, let alone with empty/declaration-only modules, so I tried to keep the behaviour as close to the current (albeit buggy) behaviour as possible.